src/status_im2/contexts/chat/composer/link_preview/events.cljs

```diff
@@ -0,0 +1,129 @@
+(ns status-im2.contexts.chat.composer.link-preview.events
+  (:require
+    [clojure.set :as set]
+    [clojure.string :as string]
+    [status-im.data-store.messages :as data-store.messages]
+    [taoensso.timbre :as log]
+    [utils.collection]
+    [utils.re-frame :as rf]))
+
+(rf/defn unfurl-urls
+  {:events [:link-preview/unfurl-urls]}
+  [{:keys [db]} text]
+  (if (string/blank? text)
+    {:db (update db :chat/link-previews dissoc :unfurled :request-id)}
+    {:json-rpc/call
+     [{:method     "wakuext_getTextURLs"
+       :params     [text]
+       :on-success #(rf/dispatch [:link-preview/unfurl-parsed-urls %])
+       :on-error   (fn [error]
+                     (log/error "Failed to parse text and extract URLs"
+                                {:error error
+                                 :event :link-preview/unfurl-urls}))}]}))
+
+(defn- urls->previews
+  [preview-cache urls]
+  (->> urls
+       (map #(get preview-cache % {:url % :loading? true}))
+       (remove :failed?)))
+
+(rf/defn unfurl-parsed-urls
+  {:events [:link-preview/unfurl-parsed-urls]}
+  [{:keys [db]} urls]
+  (let [cleared (set (get-in db [:chat/link-previews :cleared]))]
+    (when (or (not= (set urls) cleared)
+              (empty? urls))
```

Comment:
I'd swap the two conditionals, feels more natural to check if `empty` first and then if not comparing, performance wise it's probably just the same (you avoid `set` in theory, if empty).

That way you can only `set` if the non-empty (for both `cleared` and `urls`, but as far as I can see, `cleared` are already a set?

None of this really matters though, I don't think it will have any perf impact, so anything is fine

Reply:
Good point, I usually try to start with the cheapest checks first, but I forgot here.

Reply:
Done :)